app/assets/v2/js/grants/matching_funds.js

```diff
@@ -0,0 +1,235 @@
+    },
+    enableTab: function() {
+      let vm = this;
+      let urlParams = new URLSearchParams(window.location.search);
+
+      vm.tab = urlParams.get('tab');
+
+      switch (vm.tab) {
+        case 'ready_to_claim':
+          vm.tabSelected = 1;
+          break;
+        case 'claim_history':
+          vm.tabSelected = 2;
+          break;
+        default:
+          vm.tabSelected = 0;
+      }
+      window.history.replaceState({}, document.title, `${window.location.pathname}`);
+    },
+    async fetchCLRMatches() {
+      let vm = this;
+
+      this.loading = true;
+
+      // fetch clr match entries of owned grants
+      const url = '/grants/v1/api/clr-matches/';
+
+      try {
+        let result = await (await fetch(url)).json();
+
+        // update claim status + format date fields
+        result.map(async m => {
+          m.grant_payout.funding_withdrawal_date = m.grant_payout.funding_withdrawal_date
+            ? moment(m.grant_payout.funding_withdrawal_date).format('MMM D, Y')
+            : null;
+
+          m.grant_payout.grant_clrs.map(e => {
+            e.claim_start_date = e.claim_start_date ? moment(e.claim_start_date).format('MMM D') : null;
+            e.claim_end_date = e.claim_end_date ? moment(e.claim_end_date).format('MMM D, Y') : null;
+          });
+
+          const claimData = await vm.checkClaimStatus(m);
+
+          m.status = claimData.status;
+          m.claim_date = claimData.timestamp ? moment.unix(claimData.timestamp).format('MMM D, Y') : null;
+        });
+
+        // group clr matches by grant title
+        result = await this.groupByGrant(result);
+
+        // assign property for if grants have any claim or not
+        Object.keys(result).forEach(grantTitle => {
+          if (result[grantTitle].filter(a => a.has_claimed).length === 0) {
+            result[grantTitle].hasNoClaim = true;
+          } else {
+            result[grantTitle].hasNoClaim = false;
+          }
+        });
+
+        this.clrMatches = result;
+
+        this.loading = false;
+
+      } catch (e) {
+        console.error(e);
+        _alert('Something went wrong. Please try again later', 'danger');
+      }
+    },
+    async groupByGrant(clrMatches) {
+      // group clr matches by grant title
+
+      result = await clrMatches.reduce(async function(r, a) {
+        r[a.grant.title] = r[a.grant.title] || [];
+        r[a.grant.title].push(a);
+        return r;
+      }, Object.create(null));
+
+      return result;
+    },
+    async checkClaimStatus(match) {
+      const recipientAddress = match.grant.admin_address;
+      const contractAddress = match.grant_payout.contract_address;
+      const txHash = match.claim_tx;
+
+      let status = 'not-found';
+      let timestamp = null;
+
+      if (!txHash) {
+        return { status, timestamp };
+      }
+
+      web3 = new Web3(`wss://mainnet.infura.io/ws/v3/${document.contxt.INFURA_V3_PROJECT_ID}`);
```

Comment:
let's not hardcode mainnet. let's wrap it into a function where we can pass the network and get the web3. we might want to use rinkeby / matic / etc